README.md

```diff
@@ -1,72 +1,104 @@
 # icon-magic
+
 Automated icon build system for iOS, Android and Web
 
+## Glossary
+
+- **Asset**: A single file containing a logo/image. Assets can be of multiple
+  types .svg, .png, .webp
+- **Variant**: a version of the icon that has its own underlying path elements.
+  Variants are always of .svg type and form the input of the entire icon build
+  process.
+- **Icon**: The class representing a group of variants that belong to the same
+  icon. For example, two variants of the home icon can be a filled version and
+  an outline version of the same home icon. Within the file system, an icon is a
+  directory that consists of all the variant assets and a corresponding
+  iconrc.json config file.
+- **iconrc.(js|json)** A config file for a single icon or a group of icons with
+  paths to the various icon directories and their variants at minimum. The
+  config further be caustomaized
+- **Flavor**: An asset obtained after applying build/generate plugins on the
+  source .svg file. A flavor will also contain assets for each type, i. e.,
+  paths to the .svg, .png and .webp version of that flavor.
 
 ## Packages
-Icon Magic is structured as a Lerna monorepo. All of Icon Magic's packages live in the `/packages/@icon-magic` directory. A brief description of its contents is as follows:
 
+Icon Magic is structured as a Lerna monorepo. All of Icon Magic's packages live in the `/packages/@icon-magic` directory. A brief description of its contents is as follows:
 
 ### @icon-magic/blueprint
+
 A basic blueprint folder for an Icon Magic package. This module should never be published. Instead, it can be copied to another directory in `/packages/@icon-magic` (don't forget to re-name it in `package.json` too!) to create a new basic functioning module.
 
 Most Icon Magic packages are built with Typescript, tested with Mocha, and linted with tslint using custom rules from `@icon-magic/code-style`.
 
 ### @icon-magic/code-style
+
 Shared tslint config for all icon magic packages.
 
 ### @icon-magic/imagemin-farm
+
 A process farm for image minification! Has two main methods exported:
 
 #### minify(path: string): Promise\<Result>
+
 Given a path to a `png`, `jpg`, or `webp` file, minify the file. File will be modified in-place and replaced with the minified version of the file. Spins up `os.cpus() - 1` child processes to minify files. Minification tasks are transparently load balanced between processes. Promise will resolve with a `Result` object of the shape:
 
 ```typescript
 interface Result {
   path: 'path-to-file';
-  worker: number // Worker PID of completed task;
-  status: {             // Status across all tasks
-    total: number,      // Total tasks
-    remaining: number,  // Remaining tasks
-    progress: number,   // Task progress between 0 and 1.
+  worker: number; // Worker PID of completed task;
```

Comment:
Optional, but I like formatting in-line comment to all be in the same vertical line – makes code easier to skim for new collaborators! Feel free to change back, or not if you feel strongly otherwise :)

Reply:
The main README.md needs some work. I've created an issue for it. https://github.com/linkedinlabs/icon-magic/issues/7